### PR TITLE
Nix shell works with the GHC 9.2.8

### PR DESCRIPTION
**Pull request by Unisay**
- Nix build/shell uses ghc9.2.8
- HLS 1.10

### CODE REVIEW

**Review by paolino — APPROVED** [review]
Looks good to me, long time awaited step.  Can you pls wait until the benchmarks are fixed to merge ?

**Comment by Unisay** [comment]
This PR is not mergeable as windows test bundle is not buildable (caused by the Win32 core lib) without the node update.

Closing it in favor of this PR https://github.com/cardano-foundation/cardano-wallet/pull/3996 which updates both: GHC and Node (+ deps)

**Pull request closed without merging**
